### PR TITLE
fix bug 267

### PR DESCRIPTION
**Pull request by Zhaoyu1123**
Initialize Serviceinfomap whether or not local cache is loaded

### CODE REVIEW

File: clients/naming_client/naming_cache/cache/naming/DEFAULT_GROUP@@bug_267.com [hunk 1]
```diff
@@ -0,0 +1 @@
+{"dom":"","cacheMillis":10000,"useSpecifiedUrl":false,"hosts":[{"valid":false,"marked":false,"instanceId":"10.10.10.100#3600#DEFAULT#DEFAULT_GROUP@@bug_267.com","port":3600,"ip":"10.10.10.100","weight":50,"metadata":{},"clusterName":"DEFAULT","serviceName":"DEFAULT_GROUP@@bug_267.com","enabled":true,"healthy":true,"ephemeral":false}],"checksum":"","lastRefTime":1622688135226,"env":"","clusters":"","metadata":null,"name":"DEFAULT_GROUP@@bug_267.com"}
```
Comment (sanxun0325) [inline clients/naming_client/naming_cache/cache/naming/DEFAULT_GROUP@@bug_267.com:1]: unnecessary

File: clients/naming_client/naming_cache/service_info_holder.go [hunk 2]
```diff
@@ -44,9 +44,8 @@ func NewServiceInfoHolder(namespace, cacheDir string, updateCacheWhenEmpty, notL
 		subCallback:          NewSubscribeCallback(),
 		UpdateTimeMap:        cache.NewConcurrentMap(),
 	}
+	serviceInfoHolder.ServiceInfoMap = cache.NewConcurrentMap()
```
Comment (sanxun0325) [inline clients/naming_client/naming_cache/service_info_holder.go:47]: Initialize the ServiceInfoMap when you build ServiceInfoHolder，maybe it's better

File: clients/naming_client/naming_cache/service_info_holder_test.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package naming_cache
+
+import (
+    "testing"
+)
+
+func TestNewServiceInfoHolder(t *testing.T) {
+    NewServiceInfoHolder("", "./cache", true, true)
```
Comment (sanxun0325) [inline clients/naming_client/naming_cache/service_info_holder_test.go:24]: Unit tests should have assertions

**Review by sanxun0325 — CHANGES_REQUESTED** [review]

File: clients/naming_client/naming_cache/cache/naming/DEFAULT_GROUP@@bug_267.com (hunk 1 above)
Comment (Zhaoyu1123) [reply clients/naming_client/naming_cache/cache/naming/DEFAULT_GROUP@@bug_267.com:1]: If don`t add this, the test case can`t run.

File: clients/naming_client/naming_cache/service_info_holder_test.go (hunk 3 above)
Comment (Zhaoyu1123) [reply clients/naming_client/naming_cache/service_info_holder_test.go]: I had remove the test case, this seems unnecessary

File: clients/naming_client/naming_cache/service_info_holder.go [hunk 4]
```diff
@@ -43,10 +43,10 @@ func NewServiceInfoHolder(namespace, cacheDir string, updateCacheWhenEmpty, notL
 		cacheDir:             cacheDir,
 		subCallback:          NewSubscribeCallback(),
 		UpdateTimeMap:        cache.NewConcurrentMap(),
+		ServiceInfoMap:       cache.NewConcurrentMap(),
 	}
-	if notLoadCacheAtStart {
-		serviceInfoHolder.ServiceInfoMap = cache.NewConcurrentMap()
-	} else {
+	
+	if !notLoadCacheAtStart {
```
Comment (sanxun0325) [inline clients/naming_client/naming_cache/service_info_holder.go:49]: CI error, need to use gofmt format code

Comment (Zhaoyu1123) [reply clients/naming_client/naming_cache/service_info_holder.go:49]: done

**Review by sanxun0325 — APPROVED** [review]
LGTM

**Pull request merged**